core/objectUtils.js

```diff
@@ -94,6 +94,13 @@ const toIndexedObj = (array, prop) => array.reduce(
 
 const toUuidIndexedObj = R.partialRight(toIndexedObj, [keys.uuid])
 
+const indexByProps = props => array => props.reduce((acc, item, index) => {
```

Comment:
this is not an ObjectUtils method, but rather a csvParser specific issue. this can be handled directly in csvReader processNext. if there are the headers, then just return an object with key values, otherwise the array. what do you think?

Reply:
good